snuba/subscriptions/verifier.py

```diff
@@ -282,22 +283,34 @@ def _record(self, until: int) -> None:
                 tags={"outcome": "missing_from_orig"},
             )
 
-            matching_results = 0
+            matching_results: Set[str] = set()
 
             for id in intersection_ids:
                 if orig_results[id] == new_results[id]:
-                    matching_results += 1
+                    matching_results.add(id)
+
+            non_matching_results = intersection_ids - matching_results
 
             self.__metrics.increment(
-                METRIC_OUTCOME_NAME, matching_results, {"outcome": "same_result"},
+                METRIC_OUTCOME_NAME, len(matching_results), {"outcome": "same_result"},
             )
 
             self.__metrics.increment(
                 METRIC_OUTCOME_NAME,
-                len(intersection_ids) - matching_results,
+                len(non_matching_results),
                 {"outcome": "different_result"},
             )
 
+            for id in non_matching_results:
+                logger.warning(
```

Comment:
So should we have 100% matching because subscriptions only run based on matching commit offsets? Or should this perhaps be sampled because it's normal for e.g. backdated data to be inserted after tick creation but before subscription?

Reply:
There's not a lot of non matching results normally, but I'll still make a change to limit it to maybe 100 or so per second just in case.